features/tags_and_wrangling/tag_wrangling_more.feature

```diff
@@ -104,3 +104,105 @@ Feature: Tag wrangling: assigning wranglers, using the filters on the Wranglers
+        | Cowboy Bebop                           | Fandom       | true      |
+        | Faye Valentine is a sweetie            | Freeform     | false     |
+        | Ed is a sweetie                        | Freeform     | false     |
+      And I am logged in as a random user
+      And I post the work "Asteroid Blues" with fandom "Cowboy Bebop" with freeform "Ed is a sweetie"
+      And I post the work "Honky Tonk Women" with fandom "Cowboy Bebop" with freeform "Faye Valentine  is a sweetie"
+     When the tag wrangler "lain" with password "lainnial" is wrangler of "Cowboy Bebop"
+       And I follow "Tag Wrangling"
+       And I follow "2"
+       And I fill in "fandom_string" with "Cowboy Bebop"
+       And I check the mass wrangling option for "Faye Valentine is a sweetie"
+       And I check the mass wrangling option for "Ed is a sweetie"
+       And I check the canonical option for the tag "Faye Valentine is a sweetie"
+       And I check the canonical option for the tag "Ed is a sweetie"
+       And I press "Wrangle"
+     Then I should see "The following tags were successfully wrangled to Cowboy Bebop: Faye Valentine is a sweetie, Ed is a sweetie Wrangle Tags for Cowboy Bebop"
+       And the "Faye Valentine is a sweetie" tag should be canonical
+       And the "Ed is a sweetie" tag should be canonical
+
+  Scenario: Tags that don't exist cause errors
+    Given the following activated tag wrangler exists
+      | login          |
+      | wranglerette   |
+    When I am logged in as "wranglerette"
+    Then visiting "/tags/this_is_an_unknown_tag/edit" it should fail with an error
+      And visiting "/tags/this_is_an_unknown_tag/show" it should fail with an error
+      And visiting "/tags/this_is_an_unknown_tag/feed.atom" it should fail with an error
+
+  Scenario: Banned tags can only be viewed by an admin
+    Given the following typed tags exists
+        | name                                   | type         |
+        | Cowboy Bebop                           | Banned       |
+    When I am logged in as a random user
+     And I view the tag "Cowboy Bebop"
+    Then I should see "Sorry, you don't have permission to access the page you were trying to reach."
+    When I am logged in as an admin
+     And I view the tag "Cowboy Bebop"
+    Then I should not see "Please log in as an admin"
+     And I should see "Cowboy Bebop"
+
+  @javascript
+  Scenario: A user can see hidden tags
+    Given the following typed tags exists
+        | name                                   | type         | canonical |
+        | Cowboy Bebop                           | Fandom       | true      |
+        | Faye Valentine is a sweetie            | Freeform     | false     |
+        | Ed is a sweetie                        | Freeform     | false     |
+      And I am logged in as "first_user" with password "secure_password" with preferences set to hidden warnings and no additional tags
```

Comment:
Since we're not checking the behavior of hidden tags while logged in as `first_user`, this should be okay as plain old `And I am logged in as "first_user"`